edx_solutions_api_integration/courses/serializers.py

```diff
@@ -44,35 +44,42 @@ def get_completion_percentage(self, obj):
 class CourseSerializer(serializers.Serializer):
     """ Serializer for Courses """
     id = serializers.CharField()  # pylint: disable=invalid-name
-    name = serializers.CharField()
-    category = serializers.CharField()
-    number = serializers.CharField()
-    org = serializers.CharField()
+    name = serializers.CharField(source='display_name')
+    category = serializers.SerializerMethodField()
+    number = serializers.CharField(source='display_number_with_default')
+    org = serializers.CharField(source='display_org_with_default')
     uri = serializers.SerializerMethodField()
-    course_image_url = serializers.SerializerMethodField()
-    resources = serializers.SerializerMethodField()
-    due = serializers.DateTimeField()
+    course_image_url = serializers.CharField()
+    due = serializers.SerializerMethodField('get_due_date')
     start = serializers.DateTimeField()
     end = serializers.DateTimeField()
 
     def get_uri(self, course):
         """
         Builds course detail uri
         """
-        return course.get('uri', None) if isinstance(course, dict) else \
-            "{}/{}".format(generate_base_uri(self.context['request']), course.id)
-
-    def get_course_image_url(self, course):
+        request = self.context['request']
+        protocol = 'http'
```

Comment:
Can we simplify this method using `reverse` method of django rest framework? 
```
from rest_framework.reverse import reverse
```
and then
```
reverse('course-detail', args=[course.id], request=request)
```

we need to set a name(`course-detail`)  for the course detail view.